pkg/statistics/handle/syncload/stats_syncload.go

```diff
@@ -545,14 +557,18 @@ func (s *statsSyncLoad) updateCachedItem(item model.TableItemID, colHist *statis
 		}
 		tbl = tbl.Copy()
 		tbl.SetCol(item.ID, colHist)
-		// If the column is analyzed we refresh the map for the possible change.
-		if colHist.StatsAvailable() {
-			tbl.ColAndIdxExistenceMap.InsertCol(item.ID, colHist.Info, true)
-		}
+
 		// All the objects shares the same stats version. Update it here.
 		if colHist.StatsVer != statistics.Version0 {
 			tbl.StatsVer = statistics.Version0
 		}
+		if tbl.ColAndIdxExistenceMap.IsColEmpty() {
+			for _, col := range tbleInfo.Meta().Columns {
+				tbl.ColAndIdxExistenceMap.InsertCol(col.ID, col, colHist.StatsAvailable())
+			}
+		}
```

Comment:
Only update the column itself. Otherwise we need to read all column from `mysql.stats_histograms`.

Reply:
Now, it only update the column itself.